datalad/support/archive_utils_patool.py

```diff
@@ -63,6 +63,16 @@
 
 _runner = WitlessRunner()
 
+check_existing_filename = None
+check_archive_filelist = None
+try: # define check_existing_filename from patool >= 2.0.0
+    check_existing_filename = patoolib.fileutil.check_existing_filename
+    check_archive_filelist = patoolib.fileutil.check_archive_filelist
+except Exception as e: # fallback to version <2.0.0
+    check_existing_filename = patoolib.util.check_existing_filename
+    check_archive_filelist = patoolib.util.check_archive_filelist
```

Comment:
thank you @dguibert for the PR!  Here -- no need for guesswork, we have a helper to check the version, please use it instead:

```
❯ python -c 'from datalad.support.external_versions import external_versions; print(external_versions["patoolib"])'
2.2.0
❯ python -c 'from datalad.support.external_versions import external_versions; print(external_versions["patoolib"] >= "2.0")'
True
```

So I guess smth like

```suggestion
from datalad.support.external_versions import external_versions

if external_versions["patoolib"]  >= "2.0.0":
    check_existing_filename = patoolib.fileutil.check_existing_filename
    check_archive_filelist = patoolib.fileutil.check_archive_filelist
else:
    check_existing_filename = patoolib.util.check_existing_filename
    check_archive_filelist = patoolib.util.check_archive_filelist
```
?

Reply:
Thanks, I've applied your suggestion and fixed up the commit.